Package togglelog.0.1.0

WardBrian opened the pull request:
### `togglelog.0.1.0`
A ppx for compile-time-optional logging



---
* Homepage: https://github.com/WardBrian/togglelog
* Source repo: git+https://github.com/WardBrian/togglelog.git
* Bug tracker: https://github.com/WardBrian/togglelog/issues

---
:camel: Pull-request generated by opam-publish v2.2.0

mseri commented:
This is currently failing with
```
# --- a/_build/.sandbox/149194290d9462cb27e578990462a7a6/default/test/type_error.t/run.t
# +++ b/_build/.sandbox/149194290d9462cb27e578990462a7a6/default/test/type_error.t/run.t.corrected
# @@ -1,15 +1,67 @@
#  Show that a type error is thrown whether or not logging is actually enabled
#  
#    $ dune exec ./test.exe --force 2>&1 | grep "test.ml" -a5
# -  File "test.ml", line 4, characters 25-41:
# -  4 |   [%toggle_log "TESTING" (my_greeting ())]
# -                               ^^^^^^^^^^^^^^^^
# -  Error: This expression has type float but an expression was expected of type
# -           string
# +  grep: unrecognized option: 5
# +  BusyBox v1.35.0 (2022-08-01 15:14:44 UTC) multi-call binary.
# +  
# +  Usage: grep [-HhnlLoqvsrRiwFE] [-m N] [-A|B|C N] { PATTERN | -e PATTERN... | -f FILE... } [FILE]...
# +  
# +  Search for PATTERN in FILEs (or stdin)
# +  
# +  	-H	Add 'filename:' prefix
# +  	-h	Do not add 'filename:' prefix
# +  	-n	Add 'line_no:' prefix
# +  	-l	Show only names of files that match
# +  	-L	Show only names of files that don't match
# +  	-c	Show only count of matching lines
# +  	-o	Show only the matching part of line
# +  	-q	Quiet. Return 0 if PATTERN is found, 1 otherwise
# +  	-v	Select non-matching lines
# +  	-s	Suppress open and read errors
# +  	-r	Recurse
# +  	-R	Recurse and dereference symlinks
# +  	-i	Ignore case
# +  	-w	Match whole words only
# +  	-x	Match whole lines only
# +  	-F	PATTERN is a literal (not regexp)
# +  	-E	PATTERN is an extended regexp
# +  	-m N	Match up to N times per file
# +  	-A N	Print N lines of trailing context
# +  	-B N	Print N lines of leading context
# +  	-C N	Same as '-A N -B N'
# +  	-e PTRN	Pattern to match
# +  	-f FILE	Read pattern from file
# +  [2]
```

WardBrian commented:
I added the grep to try to avoid an issue where `ld` warnings were showing up in the cram test output on some platforms, but I guess the command line argument structure for grep differs sufficiently that it will cause more problems. 

Still not sure how to avoid the original ld warnings, which can be seen e.g. [here](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/7beadde5339eff16c5fc9c025066bf1c9e306567/variant/extras,x86_32-ocaml-4.14,togglelog.0.1.0,tests):

```
# File "test/type_error.t/run.t", line 1, characters 0-0:
# /usr/bin/git --no-pager diff --no-index --color=always -u _build/.sandbox/fa1ebdd9d41234d454e4014a4f0ecafb/default/test/type_error.t/run.t _build/.sandbox/fa1ebdd9d41234d454e4014a4f0ecafb/default/test/type_error.t/run.t.corrected
# diff --git a/_build/.sandbox/fa1ebdd9d41234d454e4014a4f0ecafb/default/test/type_error.t/run.t b/_build/.sandbox/fa1ebdd9d41234d454e4014a4f0ecafb/default/test/type_error.t/run.t.corrected
# index 989298a..a6cc339 100644
# --- a/_build/.sandbox/fa1ebdd9d41234d454e4014a4f0ecafb/default/test/type_error.t/run.t
# +++ b/_build/.sandbox/fa1ebdd9d41234d454e4014a4f0ecafb/default/test/type_error.t/run.t.corrected
# @@ -1,6 +1,8 @@
#  Show that a type error is thrown whether or not logging is actually enabled
#  
#    $ dune exec ./test.exe --force 2>&1 | grep "test.ml" -a5
# +  /usr/bin/ld: /home/opam/.opam/4.14/.opam-switch/build/togglelog.0.1.0/_build/install/default/lib/togglelog/togglelog.a(togglelog__Mapper.o): warning: relocation in read-only section `.text'
# +  /usr/bin/ld: warning: creating DT_TEXTREL in a PIE
#    File "test.ml", line 4, characters 25-41:
#    4 |   [%toggle_log "TESTING" (my_greeting ())]
#                                 ^^^^^^^^^^^^^^^^
# @@ -8,6 +10,8 @@ Show that a type error is thrown whether or not logging is actually enabled
#             string
#  
#    $ dune exec ./test.exe --instrument-with togglelog --force 2>&1 | grep "test.ml" -a5
# +  /usr/bin/ld: /home/opam/.opam/4.14/.opam-switch/build/togglelog.0.1.0/_build/install/default/lib/togglelog/togglelog.a(togglelog__Mapper.o): warning: relocation in read-only section `.text'
# +  /usr/bin/ld: warning: creating DT_TEXTREL in a PIE
#    File "test.ml", line 4, characters 25-41:
#    4 |   [%toggle_log "TESTING" (my_greeting ())]
# 

```

WardBrian commented on packages/togglelog/togglelog.0.1.0/opam:
```diff
@@ -0,0 +1,37 @@
+    "-j"
+    jobs
+    "@install"
+    "@runtest" {with-test}
```
What would be the equivalent to disable the tests on the `extras,x86_32-ocaml-4.14` variant?

kit-ty-kate commented:
i've reported the bug on x86_32 upstream in https://github.com/ocaml/ocaml/issues/12166. With a bit of luck it'll be fixed in the upcoming OCaml 4.14.2.

kit-ty-kate commented:
Thanks
